services/navigation/navigation.go

```diff
@@ -45,6 +60,30 @@ func (m Mode) String() string {
 	}
 }
 
+func (m MapType) String() string {
+	switch m {
+	case NoMap:
+		return "None"
+	case GPSMap:
+		return "GPS"
+	default:
+		return "UNKNOWN"
+	}
+}
+
+// StringToMapType converts an input string into one of the valid map type if possible.
+func StringToMapType(mapTypeName string) (MapType, error) {
+	switch mapTypeName {
+	case "None":
+		return NoMap, nil
+	case "GPS":
+		return GPSMap, nil
+	case "":
+		return GPSMap, nil
+	}
+	return 0, errors.Errorf("invalid map_type '%v' given", mapTypeName)
+}
+
```

Comment:
[nit]
```suggestion
	case "GPS", "":
		return GPSMap, nil
	default:
		return 0, errors.Errorf("invalid map_type '%v' given", mapTypeName)
	}
}

```

Reply:
Now that the default "" is handled by the Validate() func, I have removed it as a case. Let me know if you'd prefer I bring it back in the format you suggest above